subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/ivyservice/DefaultArtifactCaches.java

```diff
@@ -46,14 +54,35 @@ public DefaultArtifactCaches(CacheScopeMapping cacheScopeMapping, CacheRepositor
         String roCache = System.getenv(READONLY_CACHE_ENV_VAR);
         if (StringUtils.isNotEmpty(roCache)) {
             IncubationLogger.incubatingFeatureUsed("Shared read-only dependency cache");
-            readOnlyCacheMetadata = new DefaultArtifactCacheMetadata(cacheScopeMapping, new File(roCache));
-            readOnlyArtifactCacheLockingManager = new ReadOnlyArtifactCacheLockingManager(cacheRepository, readOnlyCacheMetadata);
+            File baseDir = validateReadOnlyCache(documentationRegistry, new File(roCache));
+            if (baseDir != null) {
+                readOnlyCacheMetadata = new DefaultArtifactCacheMetadata(cacheScopeMapping, baseDir);
+                readOnlyArtifactCacheLockingManager = new ReadOnlyArtifactCacheLockingManager(cacheRepository, readOnlyCacheMetadata);
+            } else {
+                readOnlyCacheMetadata = null;
+                readOnlyArtifactCacheLockingManager = null;
+            }
         } else {
             readOnlyCacheMetadata = null;
             readOnlyArtifactCacheLockingManager = null;
         }
     }
 
+    private static File validateReadOnlyCache(DocumentationRegistry documentationRegistry, File cacheDir) {
+        if (!cacheDir.exists()) {
+            LOGGER.warn("The " + READONLY_CACHE_ENV_VAR + " environment variable was set to " + cacheDir + " which doesn't exist!");
+            return null;
+        }
+        File root = CacheLayout.ROOT.getPath(cacheDir);
+        if (!root.exists()) {
+            String docLink = documentationRegistry.getDocumentationFor("dependency_resolution", "sub:shared-readonly-cache");
+            LOGGER.warn("Read-only cache is configured but the directory layout isn't expected. You must have a pre-populated " +
```

Comment:
I would add something in both messages that explicitly states that the RO cache will not be activated because of this.

Reply:
added